x-pack/apm-server/aggregation/servicemetrics/aggregator.go

```diff
@@ -184,21 +200,23 @@ func (a *Aggregator) processTransaction(event *model.APMEvent) {
 	if event.Transaction == nil || event.Transaction.RepresentativeCount <= 0 {
 		return
 	}
-	key := makeAggregationKey(event, a.config.Interval)
-	metrics := makeServiceMetrics(event)
-	a.active.storeOrUpdate(key, metrics, a.config.Interval, a.config.Logger)
+	for _, interval := range a.Intervals {
```

Comment:
I wonder if there is any point in doing this concurrently. Do we have some benchmarking data on this?

Reply:
I haven't done any benchmarking on it, but given the locking strategy and how this function is called, I fear we'd have goroutine explosion, since this is called on each batch that is processed, and we acquire the full `metricsBuffer` lock for each event we process.